serialization/untyped_boolean.go

```diff
@@ -0,0 +1,18 @@
+package serialization
+
+// UntypedBoolean defines an untyped boolean object.
+type UntypedBoolean struct {
+	UntypedNode
+}
+
+// GetValue returns the bool value.
+func (un *UntypedBoolean) GetValue() *bool {
+	return un.value.(*bool)
+}
+
+// NewUntypedBoolean creates a new UntypedBoolean object.
+func NewUntypedBoolean(boolValue *bool) *UntypedBoolean {
+	m := &UntypedBoolean{}
+	m.value = boolValue
```

Comment:
Its possible to assign a `nil` as the `boolValue` possibly constructing an untyped boolean thats an untyped null.

Reply:
Thanks for taking a look @rkodev.
Just to confirm here, what you're saying is that you can pass a nil as below?

```golang
fakeBoolValue := NewUntypedBoolean(nil)
```

And the suggestion would be to add code that guards from such scenarios?

Reply:
Yes, we should throw an error when a nil value is passed for as the argument for a constructor